src/core/geometry/qgslinestring.cpp

```diff
@@ -187,6 +187,90 @@ bool QgsLineString::isEmpty() const
   return mX.isEmpty();
 }
 
+QgsLineString *QgsLineString::asGridified( double hSpacing, double vSpacing, double dSpacing, double mSpacing,
+    double /*tolerance*/, SegmentationToleranceType /*toleranceType*/ ) const
+{
+  int length = numPoints();
+
+  if ( length <= 0 )
+    return nullptr;
+
+
+  QgsLineString *result;
```

Comment:
unique_ptr here for result

Reply:
It is the final result (the pointer which is returned)